diff --git a/pydantic/aliases.py b/pydantic/aliases.py
--- a/pydantic/aliases.py
+++ b/pydantic/aliases.py
@@ -113,11 +113,9 @@
             TypeError: If the alias generator produces an invalid type.
         """
         alias = None
-        alias_generator = getattr(self, alias_kind)
-
-        if alias_generator is not None:
+        if alias_generator := getattr(self, alias_kind):
             alias = alias_generator(field_name)
-            if alias is not None and not isinstance(alias, allowed_types):
+            if alias and not isinstance(alias, allowed_types):
                 raise TypeError(
                     f'Invalid `{alias_kind}` type. `{alias_kind}` generator must produce one of `{allowed_types}`'
                 )

